### PR TITLE
Add `transform-regexp-modifiers` to `preset-env`

### PR DESCRIPTION
**Pull request by JLHwung**
| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | 
| Patch: Bug Fix?          |
| Major: Breaking Change?  |
| Minor: New Feature?      | Yes
| Tests Added + Pass?      | Yes
| Documentation PR Link    | https://github.com/babel/website/pull/2977
| Any Dependency Changes?  |
| License                  | MIT

Added the `regexp-modifiers` plugin to `preset-env` now that it just reaches Stage 4 in the TC39 Oct 2024 meeting.

### CODE REVIEW

**Review by liuxingbaoyu — APPROVED** [review]

**Review by nicolo-ribaudo — APPROVED** [review]

**Comment by nicolo-ribaudo** [comment]
@JLHwung This is on the agenda for stage 4 at the TC39 meeting next week.

**Comment by JLHwung** [comment]
Thanks. I will update this PR and rename the plugin if it reaches stage 4.

**Comment by nicolo-ribaudo** [comment]
This reached consensus for stage 4.

The extra change proposed by @JLHwung did not reach consensus, as the proposal has already been implemented as-is and implementations didn't consider it to be worth the change at this point.

File: .github/workflows/reminders.yml [hunk 1]
```diff
@@ -20,4 +20,5 @@ jobs:
             Please run these commands:
             ```
             npm deprecate @babel/plugin-proposal-duplicate-named-capturing-groups-regex "This proposal has been merged to the ECMAScript standard and thus this plugin is no longer maintained. Please use @babel/plugin-transform-duplicate-named-capturing-groups-regex instead."
+            npm deprecate @babel/plugin-proposal-regexp-modifiers "This proposal has been merged to the ECMAScript standard and thus this plugin is no longer maintained. Please use @babel/plugin-transform-regexp-modifiers instead."
```
Comment (nicolo-ribaudo) [inline .github/workflows/reminders.yml:23]: Can you update the cronjob schedule to be at the end of november?

File: packages/babel-preset-env/src/available-plugins.ts [hunk 2]
```diff
@@ -153,6 +155,7 @@ if (!process.env.BABEL_8_BREAKING) {
     "bugfix/transform-safari-id-destructuring-collision-in-function-expression":
       "7.16.0",
     "bugfix/transform-v8-static-class-fields-redefine-readonly": "7.12.0",
+    "proposal-regexp-modifiers": "7.19.0",
```
Comment (nicolo-ribaudo) [inline packages/babel-preset-env/src/available-plugins.ts:158]: Should this be transform-?

**Review by liuxingbaoyu — APPROVED** [review]

**Comment by JLHwung** [comment]
Rebased due to compat-data updates from Node 23 and Firefox 132.

**Pull request merged**